ddtrace/contrib/asgi/middleware.py

```diff
@@ -128,12 +128,21 @@ async def __call__(self, scope, receive, send):
         if sample_rate is not None:
             span.set_tag(ANALYTICS_SAMPLE_RATE_KEY, sample_rate)
 
+        host_header = None
+        for key, value in scope["headers"]:
+            if key == b"host":
+                host_header = value.decode("latin-1")
```

Comment:
do we need a try/catch here? we shouldn't have any UTF-8 in domain names? 🤔

Reply:
I'll just use stringify instead. I am not sure why we would need to decode to `latin-1` here

Reply:
Looking into this all [uwsgi headers](https://asgi.readthedocs.io/en/latest/specs/www.html#wsgi-encoding-differences) should contain `latin1` supported characters but asgi does not have this same restriction (it just specifies byte strings and forwards whatever encoding is sent by the client to the server). It seems like quite a few libraries just assume latin-1 encoding for asgi request headers (ex: [startlete](https://github.com/encode/starlette/blob/e7a92ee4cd43b6d7acf4e2fdedb20fe0e86f9620/starlette/datastructures.py#L34) and [django](https://github.com/django/django/blob/main/django/core/handlers/asgi.py#L93) so I am not sure if I am missing something. I'll just add error handling here to be safe.